### PR TITLE
stream: add exception policy for midstream flows [6.0.x backports]- v1

### PR DESCRIPTION
**Pull request by jufajardini**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
Exception policy midstream
- https://redmine.openinfosecfoundation.org/issues/5500
applayer parser sections merging:
- https://redmine.openinfosecfoundation.org/issues/5364
starts to address documenting the exception policy:
- https://redmine.openinfosecfoundation.org/issues/5475

Describe changes:
- add a stream.midstream-policy option that allows for controlling if a flow should be fully dropped, passed or bypassed. It is also possible to just ignore the policy (keeps normal behavior) if it is a midstream flow (in IPS mode)
- merge the two existing application layer parser configuration sections in the userguide into one
- add a very brief mention to each of the existing exception policy options (that are configurable via suricata.yaml) to the userguide

suricata-verify-pr: 921
https://github.com/OISF/suricata-verify/pull/921

### CODE REVIEW

File: doc/userguide/configuration/suricata-yaml.rst [hunk 1]
```diff
@@ -1572,6 +1601,140 @@ the limits are exceeded, and an event will be raised.
 `max-write-queue-size` and `max-write-queue-cnt` are as the READ variants,
 but then for WRITEs.
 
+Configure HTTP2
```
Comment (victorjulien) [inline doc/userguide/configuration/suricata-yaml.rst:1604]: this is new here, is it intended?

File: doc/userguide/configuration/suricata-yaml.rst [hunk 2]
```diff
@@ -1572,6 +1601,140 @@ the limits are exceeded, and an event will be raised.
...
+incompatible with ``decode-mime``. If both are enabled,
+``raw-extraction`` will be automatically disabled.
+
+::
+
+      smtp:
+        # extract messages in raw format from SMTP
+        raw-extraction: true
+
+Maximum transactions
```
Comment (victorjulien) [inline doc/userguide/configuration/suricata-yaml.rst:1729]: master only?

Comment (jufajardini) [reply doc/userguide/configuration/suricata-yaml.rst:1729]: 🤐  I didn't know that, sorry. And thanks.

File: doc/userguide/configuration/suricata-yaml.rst (hunk 1 above)
Comment (jufajardini) [reply doc/userguide/configuration/suricata-yaml.rst:1604]: 🤔  👀  I'll check what happened.

Comment (jufajardini) [reply doc/userguide/configuration/suricata-yaml.rst:1604]: I think that when I merged the two sections git interpreted that it was a new addition from the commit.

From what I could see, the settings mentioned here exist in six, though.

Should I add them as a separate commit?

Comment (victorjulien) [reply doc/userguide/configuration/suricata-yaml.rst:1604]: I think that makes sense, ya.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 8957

**Comment by jufajardini** [comment]
Replaced by: https://github.com/OISF/suricata/pull/7817

**Pull request closed without merging**